### PR TITLE
Tests for gmf + tests for gmf's themesservice

### PR DESCRIPTION
**Pull request by ger-benjamin**
- Manage tests in ``contribs/gmf/test/...`` (commit 	*Add test in gmf*)
 - Make an example with ``contribs/gmf/src/services/themesservice.js`` (commit 	*Tests for gmf themesservice*)

I don't know if I should make two distinct "karma" files ..?

### CODE REVIEW

File: contribs/gmf/src/services/themesservice.js [hunk 1]
```diff
@@ -54,13 +58,13 @@ gmf.Themes = function($http, treeUrl, isThemePrivateUrl) {
...
    */
-  this.treeUrl_ = treeUrl;
+  this.treeUrl_ = '';
 
   /**
    * @type {string}
    * @private
    */
-  this.isThemePrivateUrl_ = isThemePrivateUrl;
+  this.isThemePrivateUrl_ = '';
```
Comment (fgravin) [inline contribs/gmf/src/services/themesservice.js]: `isThemePrivateUrl ` is useless, you can remove it

Comment (sbrunner) [inline contribs/gmf/src/services/themesservice.js]: +1 to remove it :-)

File: contribs/gmf/src/services/themesservice.js [hunk 2]
```diff
@@ -72,6 +76,21 @@ goog.inherits(gmf.Themes, goog.events.EventTarget);
 
 
 /**
+ * Init the service.
+ * @param {string} treeUrl URL to "themes" web service.
+ * @param {string} isThemePrivateUrl URL to check if theme is public.
+ * @export
+ */
+gmf.Themes.prototype.init = function(treeUrl, isThemePrivateUrl) {
```
Comment (ger-benjamin) [inline contribs/gmf/src/services/themesservice.js]: I've changed how to initialize the service (I use an init function to be coherent with others services) but that causes trouble with the Luxembourg project... Must I revert this change ?

File: contribs/gmf/src/services/themesservice.js (hunk 1 above)
Comment (ger-benjamin) [inline contribs/gmf/src/services/themesservice.js]: Okay... then I pass the URL as parameter on function 'isThemePrivate' ? Luxembourg project will must be adapted.

Comment (fgravin) [inline contribs/gmf/src/services/themesservice.js]: Luxembourg doesn't use this service yet.
If gmf2 doesn't support private theme url, then we need to remove it.
If a project like luxembourg want to use it, they need to add custom code in their own sources to handle it.

**Comment by fgravin** [comment]
Ok for the change of the `ThemeService`, we need to go ahead with this cause it's used by many other components though.

Thanks for the tests it's really helpfull.

File: contribs/gmf/src/services/themesservice.js (hunk 2 above)
Comment (fgravin) [inline contribs/gmf/src/services/themesservice.js]: Luxembourg has its own service for the moment, when they want to migrate to use the ngeo one, then we'll see what are the impacts. They probably need to update their application code anyway.

**Comment by ger-benjamin** [comment]
Thanks @fgravin 
So, I've removed the "isThemePrivate" function (and the corresponding test) and "isThemePrivateURL" string.

**Comment by ger-benjamin** [comment]
Can I merge ?

**Comment by fgravin** [comment]
Yes go ahead, thanks
